server/src/main/java/com/google/android/attestation/ParsedAttestationRecord.java

```diff
@@ -69,6 +70,48 @@ public abstract class ParsedAttestationRecord {
   @SuppressWarnings("Immutable")
   public abstract PublicKey attestedKey();
 
+  public static Builder builder() {
+    return new AutoValue_ParsedAttestationRecord.Builder()
+        .setAttestationChallenge(ByteString.EMPTY)
+        .setUniqueId(ByteString.EMPTY)
+        .setSoftwareEnforced(AuthorizationList.builder().build())
+        .setTeeEnforced(AuthorizationList.builder().build());
+  }
+
+  /** Builder for {@link ParsedAttestationRecord}. */
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract Builder setAttestationVersion(int value);
+
+    public abstract Builder setAttestationSecurityLevel(SecurityLevel value);
+
+    public abstract Builder setKeymasterVersion(int value);
+
+    public abstract Builder setKeymasterSecurityLevel(SecurityLevel value);
+
+    public abstract Builder setAttestationChallenge(ByteString value);
+
+    @CanIgnoreReturnValue
+    public final Builder setAttestationChallenge(byte[] value) {
+      return setAttestationChallenge(ByteString.copyFrom(value));
+    }
+
+    public abstract Builder setUniqueId(ByteString value);
+
+    @CanIgnoreReturnValue
+    public final Builder setUniqueId(byte[] value) {
+      return setUniqueId(ByteString.copyFrom(value));
+    }
+
+    public abstract Builder setSoftwareEnforced(AuthorizationList value);
```

Comment:
bikeshed nit: `setSoftwareEnforcedTags` and same for `setTeeEnforced`

Reply:
https://developer.android.com/privacy-and-security/security-key-attestation#key_attestation_ext_schema

The method names are currently faithfully transcribed from the ASN.1 schema.

Reply:
Fair enough